packages/libs/eda/src/lib/core/components/filter/HistogramFilter.tsx

```diff
@@ -654,122 +656,148 @@ function HistogramPlotWithControls({
       />
 
       <div style={{ display: 'flex', flexDirection: 'row' }}>
-        <LabelledGroup label="X-axis controls">
-          <BinWidthControl
-            binWidth={data?.binWidthSlider?.binWidth}
-            binWidthStep={data?.binWidthSlider?.binWidthStep}
-            binWidthRange={data?.binWidthSlider?.binWidthRange}
-            binUnit={uiState.binWidthTimeUnit ?? 'year'}
-            binUnitOptions={
-              data?.binWidthSlider?.valueType === 'date'
-                ? ['day', 'week', 'month', 'year']
-                : undefined
-            }
-            onBinWidthChange={handleBinWidthChange}
-            valueType={data?.binWidthSlider?.valueType}
-            containerStyles={{ minHeight: widgetHeight }}
-          />
-
-          <AxisRangeControl
-            label="Range"
-            range={uiState.independentAxisRange}
-            onRangeChange={handleIndependentAxisRangeChange}
-            valueType={data?.binWidthSlider?.valueType}
-            containerStyles={{ minWidth: '400px' }}
-          />
-          {/* truncation notification */}
-          {truncatedIndependentAxisWarning ? (
-            <Notification
-              title={''}
-              text={truncatedIndependentAxisWarning}
-              // this was defined as LIGHT_BLUE
-              color={'#5586BE'}
-              onAcknowledgement={() => {
-                setTruncatedIndependentAxisWarning('');
-              }}
-              showWarningIcon={true}
-              containerStyles={{
-                maxWidth:
-                  data?.binWidthSlider?.valueType === 'date'
-                    ? '34.5em'
-                    : '38.5em',
-              }}
-            />
-          ) : null}
-          <Button
-            type={'outlined'}
-            text={'Reset X-axis to defaults'}
-            onClick={handleIndependentAxisSettingsReset}
-            containerStyles={{
-              paddingTop: '1.0em',
-              width: '50%',
-              float: 'right',
+        <div style={{ display: 'flex', flexDirection: 'column' }}>
+          {/* set Undo icon and its behavior */}
+          <div
+            style={{
+              display: 'flex',
+              flexDirection: 'row',
+              alignItems: 'center',
             }}
-          />
-        </LabelledGroup>
+          >
+            <LabelledGroup label="X-axis controls"> </LabelledGroup>
+            <div style={{ marginLeft: '-2.6em', width: '50%' }}>
+              <ResetButtonCoreUI
+                size={'medium'}
+                text={''}
+                themeRole={'primary'}
+                tooltip={'Reset to defaults'}
+                disabled={false}
+                onPress={handleIndependentAxisSettingsReset}
```

Comment:
does this function cause the loading icon to show up? Every time i press the button i see the little spinner friend atop the plot.

Reply:
Hi @asizemore Thank you for your tests and reviews. I have tested this and found that it even happened on old button style as well (tested using qa site) so it looks like it existed quite a long time without notice. I think that this deserves another ticket to be addressed.